### PR TITLE
Annotations router custom action pre-format callback

### PR DESCRIPTION
**Pull request by BeMySlaveDarlin**
Hello!

* Type: new feature

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Added method to set callback handler to pre-format action name in Annotation Router.
Default behavior of action name processing:
`PostsController::getLastUpdatedAction()` equals to uri `/posts/getlastupdated` via strtolower formatting.
New feature allows to set callback, which does user defined pre-format to action name:
`PostsController::getLastUpdatedAction()` equals to uri `/posts/get-last-updated` via uncamelize passed as callback

Thanks

### CODE REVIEW

**Review by sergeyklay — APPROVED** [review]
There are not related failures (due to PHP setup on Travis). But in general looks good to me

**Review by ruudboon — APPROVED** [review]

**Comment by ruudboon** [comment]
Looks good to me. @BeMySlaveDarlin thnx

**Review by Jeckerson — APPROVED** [review]

**Comment by BeMySlaveDarlin** [comment]
> There are not related failures (due to PHP setup on Travis). But in general looks good to me

Fixed all issues, build successfull

**Comment by ruudboon** [comment]
Can you rebase your pull @BeMySlaveDarlin

File: .gitignore [hunk 1]
```diff
@@ -50,6 +50,7 @@ autom4te.cache/
 
 .zephir/
 .temp/
+.idea/
```
Comment (ruudboon) [inline .gitignore]: please set solve your git ignore locally (global). See: https://github.com/phalcon/cphalcon/pull/12572

Comment (BeMySlaveDarlin) [reply .gitignore]: Done

**Comment by BeMySlaveDarlin** [comment]
> Can you rebase your pull @BeMySlaveDarlin

Done

File: .travis.yml [hunk 2]
```diff
@@ -90,6 +90,7 @@ script:
   - vendor/bin/codecept build --quiet
   - vendor/bin/codecept run --ext DotReporter tests/cli/
   - vendor/bin/codecept run --ext DotReporter tests/integration/
+  - vendor/bin/codecept run --ext DotReporter tests/integration/
```
Comment (ruudboon) [inline .travis.yml]: I think this line can go.

File: tests/tocheck-database/Mvc/Model/Resultset/GetFirstCest.php [hunk 3]
```diff
@@ -41,7 +43,11 @@ public function mvcModelResultsetGetFirst(IntegrationTester $I)
         /**
          * Setup the table
          */
-        (new ObjectsMigration())($this->container->get('db'));
+        $oMigration = new ObjectsMigration($this->container->get('db'));
```
Comment (ruudboon) [inline tests/tocheck-database/Mvc/Model/Resultset/GetFirstCest.php]: Can you use full name like $objectMigration and $stuffMigration looks like an Hungarian notation now

**Review by ruudboon — CHANGES_REQUESTED** [review]

File: tests/tocheck-database/Mvc/Model/Resultset/GetFirstCest.php (hunk 3 above)
Comment (niden) [reply tests/tocheck-database/Mvc/Model/Resultset/GetFirstCest.php]: @ruudboon we can leave this here, these tests needs to be rechecked after the refactoring with the database suite so we can correct it when we start moving stuff around.

**Review by niden — APPROVED** [review]

File: tests/tocheck-database/Mvc/Model/Resultset/GetFirstCest.php (hunk 3 above)
Comment (ruudboon) [reply tests/tocheck-database/Mvc/Model/Resultset/GetFirstCest.php]: +1

**Comment by niden** [comment]
Actually, there is a mixup with the rebase. The tests fail because there are two declarations of `insert` within the `InvoicesMigration`. @BeMySlaveDarlin Do you mind rebasing again and having a look on where the duplication is?

**Comment by BeMySlaveDarlin** [comment]
Messed up fork with rebase over 67 commits, recreating PR, sorry =)

**Pull request closed without merging**